cpp/include/cugraph/algorithms.hpp

```diff
@@ -1677,6 +1679,99 @@ void triangle_count(raft::handle_t const& handle,
                     raft::device_span<edge_t> counts,
                     bool do_expensive_check = false);
 
+/**
+ * @brief     Compute Jaccard similarity coefficient
+ *
+ * Similarity is computed based for every pair of vertices specified, or (if no vertices
+ * are specified then all vertices that are 2 hops apart are computed).  Note that
+ * similarity algorithms expect a symmetric graph.
+ *
+ * @throws                 cugraph::logic_error when an error occurs.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Graph view object.
+ * @param first optional device span defining the first vertex in the desired vertex pairs
+ * @param second optional device span defining the second vertex in the desired vertex pairs
+ * @param use_weights If true use the weights associated with the graph.  If false assume a weight
+ *                    of 1 for all edges.
+ * @return tuple defining vertex pairs and their Jaccard similarity coefficient.
```

Comment:
Should we return the vertex pairs as well? Or should this be optional and vertex pairs should be returned only when `first` and `second` are not provided?

And if we provide a function to easily compute two-hop neighbors, we may make `first` and `second` required parameters. If uses want to compute coefficients for two-hop neighbors, they call the function to compute two-hop neighbors first and call this function.

Reply:
I did it this way largely for this discussion :-)

The legacy implementation had a function to compute two-hop neighbors and then made this a required field.  We could mimic that behavior here.  I do think that's a bit cleaner.  I provided this definition as an alternative... essentially we embed the computation of the two-hop neighbors in the function.

If you're on board with the change, I'm happy to
* add a function to compute two-hop neighbors
* Make the vertex_pairs parameter required
* Only return the scores, ordered to match the vertex_pairs

Reply:
Yes, I am on board. I think the suggested design is much more composable and also better stick to the one-function-does-one-thing idea.

Reply:
I will put it in the next push.